packages/agents-hosting/src/auth/msalConnectionManager.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Activity, RoleTypes } from '@microsoft/agents-activity'
+import { AuthConfiguration } from './authConfiguration'
+import { AuthProvider } from './authProvider'
+import { Connections } from './connections'
+import { MsalTokenProvider } from './msalTokenProvider'
+
+export interface ConnectionMapItem {
+  audience?: string
+  serviceUrl: string
+  connection: string
+}
+
+export class MsalConnectionManager implements Connections {
+  private _connections: Map<string, MsalTokenProvider>
+  private _connectionsMap: ConnectionMapItem[]
+  private _serviceConnectionConfiguration: AuthConfiguration
+  private static readonly DEFAULT_CONNECTION = 'serviceConnection'
+
+  constructor (
+    connectionsConfigurations: Map<string, AuthConfiguration> = new Map(),
+    connectionsMap: ConnectionMapItem[] = [],
+    configuration: AuthConfiguration = {}) {
+    this._connections = new Map()
+    this._connectionsMap = connectionsMap.length > 0 ? connectionsMap : (configuration.connectionsMap || [])
+    this._serviceConnectionConfiguration = {}
+
+    const providedConnections = connectionsConfigurations.size > 0 ? connectionsConfigurations : (configuration.connections || new Map())
+
+    for (const [name, config] of providedConnections) {
+      // Instantiate MsalTokenProvider for each connection
+      this._connections.set(name, new MsalTokenProvider(config))
+      if (name === MsalConnectionManager.DEFAULT_CONNECTION) {
+        this._serviceConnectionConfiguration = config
+      }
+    }
+
+    if (!this._connections.has(MsalConnectionManager.DEFAULT_CONNECTION)) {
```

Comment:
We will need to change this.  There should be no expectation of a "ServiceConnection".  Python has the same issue.  This will also impact `getDefaultConnection`.